File: Chatgpt/Java/2.java
import java.nio.charset.StandardCharsets;
import java.nio.file.*;

public class SafeFileLoader {
    private static final Path BASE_DIR = Paths.get("safe_files").toAbsolutePath().normalize();

    public static void main(String[] args) {
        try {
            Files.createDirectories(BASE_DIR);
            Path sample = BASE_DIR.resolve("example.txt");
            if (!Files.exists(sample)) {
                Files.writeString(sample, "This is a safe example file.\n", StandardCharsets.UTF_8);
            }

            String fileName = args.length > 0 ? args[0] : "example.txt";
            Path target = BASE_DIR.resolve(fileName).normalize();

            if (!target.startsWith(BASE_DIR)) {
                System.out.println("Access denied.");
                return;
            }

            if (!Files.exists(target) || !Files.isRegularFile(target)) {
                System.out.println("File not found.");
                return;
            }

            String content = Files.readString(target, StandardCharsets.UTF_8);
            System.out.println("File content:");
            System.out.println(content);

        } catch (Exception e) {
            System.out.println("Error reading file.");
        }
    }
}

File: Chatgpt/Python/2.py
from flask import Flask, request, jsonify
from pathlib import Path

app = Flask(__name__)

BASE_DIR = Path("safe_files").resolve()
BASE_DIR.mkdir(exist_ok=True)

sample_file = BASE_DIR / "example.txt"
if not sample_file.exists():
    sample_file.write_text("This is a safe example file.\n", encoding="utf-8")

@app.route("/read", methods=["GET"])
def read_file():
    filename = request.args.get("file", "").strip()

    if not filename:
        return jsonify({"error": "Missing file parameter"}), 400

    target = (BASE_DIR / filename).resolve()

    if not str(target).startswith(str(BASE_DIR)):
        return jsonify({"error": "Access denied"}), 403

    if not target.is_file():
        return jsonify({"error": "File not found"}), 404

    try:
        content = target.read_text(encoding="utf-8")
        return jsonify({"file": filename, "content": content})
    except Exception:
        return jsonify({"error": "Could not read file"}), 500

if __name__ == "__main__":
    app.run(debug=True)
